### ISSUE TITLE
a11y: Non-descriptive link text for `Show/Hide` Password area button

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#14123]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

`a11y issue 47`

The link text `Show/Hide` in the `Password` component does not describe the destination it points to, making it difficult for screen reader users to understand the purpose of the link effectively.

- Login
- Change Password
- User: Create
Main Content
- Show 
- Hide (available after activating Show)

**To Reproduce**

**Result**
The link text does not describe the destination it points to, making it difficult for screen reader users to understand the purpose of the link effectively.

**Expected Result**

Link text should be meaningful and unique. Add a descriptive label to the link/button

**Screenshots**

**Additional context**

**Comment by izaac** [comment]
Rancher version: `v2.12-039e8cf6a4ca96c00a271f12e05144dd0a8d4a7a-head`
Rancher UI version (dashboard): `master f5305ed`

The aria text uses a synonym `Show/Reveal` for the Show link/button which is acceptable. Other than that the functionality works as expected using Chrome/Firefox and MacOS VoiceOver.
